### PR TITLE
Add ability to create PostgreSQL backed GeoGIG repositories via REST API

### PR DESCRIPTION
**Pull request by emerkle826**
- Allow for specifying repository config form elements
- Update .gitignore to ignore generated Cucumber test reports

### CODE REVIEW

File: src/community/geogig/src/main/java/org/geogig/geoserver/config/RepositoryInfo.java [hunk 1]
```diff
@@ -84,14 +92,29 @@ void setName(String name) {
...
+            return repoName;
+        }
+        if (this.location != null) {
+            try {
+                // lookup the resolver
+                RepositoryResolver resolver = RepositoryResolver.lookup(this.location);
+                // if the repo exists, get the name from it
+                if (resolver.repoExists(this.location)) {
+                    // it exists, load it and fetch the name
+                    Repository repo = RepositoryResolver.load(this.location);
```
Comment (groldan) [inline src/community/geogig/src/main/java/org/geogig/geoserver/config/RepositoryInfo.java]: looks like a `repo.close()` wouldn't hurt after this. Actually would be a desirable thing

```
                    Repository repo = RepositoryResolver.load(this.location);
                    try {
                        return repo.command(ResolveRepositoryName.class).call();
                    } finally {
                        repo.close();
                    }
```

File: src/community/geogig/src/main/java/org/geogig/geoserver/web/repository/RepositoryEditPanel.java [hunk 2]
```diff
@@ -67,8 +70,13 @@ public void validate(IValidatable<RepositoryInfo> validatable) {
...
                     try {
-                        if (resolver.repoExists(location)) {
-                            error.addMessageKey("errRepositoryExists");
+                        if (isNew) {
+                            if (resolver.repoExists(location)) {
+                                error.addMessageKey("errRepositoryExists");
+                            }
+                        } else {
+                            // try to connect
+                            resolver.open(location);
```
Comment (emerkle826) [inline src/community/geogig/src/main/java/org/geogig/geoserver/web/repository/RepositoryEditPanel.java]: @groldan I'm guessing I need to be closing the repository here as well

File: src/community/geogig/src/main/java/org/geogig/geoserver/config/RepositoryInfo.java (hunk 1 above)
Comment (emerkle826) [inline src/community/geogig/src/main/java/org/geogig/geoserver/config/RepositoryInfo.java]: fixed

File: src/community/geogig/src/main/java/org/geogig/geoserver/web/repository/RepositoryEditPanel.java (hunk 2 above)
Comment (emerkle826) [inline src/community/geogig/src/main/java/org/geogig/geoserver/web/repository/RepositoryEditPanel.java]: fixed.

**Pull request merged**